src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/PooledObjects/PooledArrayBuilder`1.cs

```diff
@@ -2,195 +2,83 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Buffers;
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 using Microsoft.AspNetCore.Razor.Utilities;
-using Microsoft.Extensions.ObjectPool;
 
 namespace Microsoft.AspNetCore.Razor.PooledObjects;
 
 /// <summary>
-///  Wraps a pooled <see cref="ImmutableArray{T}.Builder"/> but doesn't allocate it until
-///  it's needed. Note: Dispose this to ensure that the pooled array builder is returned
-///  to the pool.
-///
-///  There is significant effort to avoid retrieving the <see cref="ImmutableArray{T}.Builder"/>.
-///  For very small arrays of length 4 or less, the elements will be stored on the stack. If the array
-///  grows larger than 4 elements, a builder will be employed. Afterward, the build will
-///  continue to be used, even if the arrays shrinks and has fewer than 4 elements.
+///  Wraps a pooled array but doesn't allocate it until it's needed. Provides
+///  access to the backing array as a <see cref="Span{T}"/>.
+///  Note: Disposal ensures the pooled array is returned to the pool.
 /// </summary>
 [NonCopyable]
```

Comment:
Is it worth adding some notes about capacity behaviour? ie adding could resize and cause an array copy, and that nothing fancy is done to prevent LOH etc. so there are better choices for really big arrays. Though I might just be being a bit paranoid :)